examples/vpc-peering-common/main.tf

```
@@ -0,0 +1,200 @@
+
+  peer_export_custom_routes                = each.value.peer_export_custom_routes
+  peer_import_custom_routes                = each.value.peer_import_custom_routes
+  peer_export_subnet_routes_with_public_ip = each.value.peer_export_subnet_routes_with_public_ip
+  peer_import_subnet_routes_with_public_ip = each.value.peer_import_subnet_routes_with_public_ip
+}
+
+module "vmseries" {
+  source = "../../modules/vmseries"
+
+  for_each = var.vmseries
+
+  name                  = "${var.name_prefix}${each.value.name}"
+  zone                  = each.value.zone
+  ssh_keys              = var.vmseries_common.ssh_keys
+  vmseries_image        = var.vmseries_common.vmseries_image
+  machine_type          = each.value.machine_type
+  min_cpu_platform      = try(each.value.min_cpu_platform, "Intel Cascade Lake")
+  tags                  = try(each.value.tags, [])
+  service_account       = module.iam_service_account[each.value.service_account].email
+  scopes                = try(each.value.scopes, [])
+  create_instance_group = true
+
+  bootstrap_options = try(merge({ vmseries-bootstrap-gce-storagebucket = "${module.bootstrap[each.value.bootstrap-bucket-key].bucket_name}/${each.key}/" }, var.vmseries_common.bootstrap_options), merge(
+    try(each.value.bootstrap_options, {}),
+    try(var.vmseries_common.bootstrap_options, {})
+  ))
+
+  named_ports = try([for v in each.value.named_ports :
+    {
+      name = v.name
+      port = v.port
+  }], [])
+
+  network_interfaces = [for v in each.value.network_interfaces :
+    {
+      subnetwork       = module.vpc.subnetworks["${var.name_prefix}${v.subnetwork}"].self_link
+      private_ip       = v.private_ip
+      create_public_ip = try(v.create_public_ip, false)
+  }]
+}
+
+data "google_compute_image" "my_image" {
+  family  = "ubuntu-pro-2204-lts"
+  project = "ubuntu-os-pro-cloud"
+}
+
+resource "google_compute_instance" "linux_vm" {
+  for_each = var.linux_vms
+
+  name         = "${var.name_prefix}${each.key}"
+  machine_type = each.value.linux_machine_type
+  zone         = each.value.zone
+
+  boot_disk {
+    initialize_params {
+      image = data.google_compute_image.my_image.id
+      size  = each.value.linux_disk_size
+    }
+  }
+
+  network_interface {
+    subnetwork = module.vpc.subnetworks["${var.name_prefix}${each.value.subnetwork}"].self_link
+    network_ip = each.value.private_ip
+  }
+
+  metadata = {
+    ssh-keys       = each.value.ssh_keys_linux
+    enable-oslogin = true
+  }
+
+
+  service_account {
+    email  = module.iam_service_account[each.value.service_account].email
+    scopes = each.value.scopes
+  }
+}
+
+module "lb_internal" {
+  source = "../../modules/lb_internal"
+
+  for_each = var.lbs_internal
+
+  name              = "${var.name_prefix}${each.value.name}"
+  health_check_port = try(each.value.health_check_port, "80")
+  backends          = { for v in each.value.backends : v => module.vmseries[v].instance_group_self_link }
```

```suggestion
  backends          = { for k, v in module.vmseries : k => v.instance_group_self_link }
```
Then ilb backends parameter becomes unnecessary.

---

Code allows for flexibility when dealing with different vm-series calls and while calling the module `vmseries` a single time for both dedicated inbound and common examples model.